pelican/writers.py

```diff
@@ -143,11 +143,15 @@ def write_feed(self, elements, context, path=None, url=None,
 
         feed = self._create_new_feed(feed_type, feed_title, context)
 
-        max_items = len(elements)
         if self.settings['FEED_MAX_ITEMS']:
-            max_items = min(self.settings['FEED_MAX_ITEMS'], max_items)
-        for i in range(max_items):
-            self._add_item_to_the_feed(feed, elements[i])
+            last_item = self.settings['FEED_MAX_ITEMS']
+        elif isinstance(self.settings['FEED_MAX_ITEMS'], bool):
+            # explicitly set FEED_MAX_ITEMS = False
+            last_item = len(elements)
+        else:
+            last_item = 100
```

Comment:
Default should go into [`settings.py`](https://github.com/getpelican/pelican/blob/86f62d0a92ad78df36aac9f5837d4f2715535421/pelican/settings.py#L43)